charts/nextcloud/templates/deployment.yaml

```diff
@@ -308,23 +326,23 @@ spec:
       {{- if .Values.nextcloud.extraVolumes }}
 {{ toYaml .Values.nextcloud.extraVolumes | indent 6 }}
       {{- end }}
-      {{- if .Values.nginx.enabled }}
-      # Will mount configuration files as www-data (id: 82) for nextcloud
       securityContext:
-        fsGroup: 82
-        {{- if .Values.securityContext }}
-        {{- with .Values.securityContext }}
-        {{- toYaml . | nindent 8 }}
-        {{- end }}
+      {{- if .Values.nextcloud.podSecurityContext }}
+        {{- with .Values.nextcloud.podSecurityContext }}
+          {{- toYaml . | nindent 8 }}
         {{- end }}
       {{- else }}
-      # Will mount configuration files as www-data (id: 33) for nextcloud
-      securityContext:
+        {{- if .Values.nginx.enabled }}
+        # Will mount configuration files as www-data (id: 82) for nextcloud
+        fsGroup: 82
+        {{- else }}
+        # Will mount configuration files as www-data (id: 33) for nextcloud
         fsGroup: 33
-        {{- if .Values.securityContext }}
-        {{- with .Values.securityContext }}
-        {{- toYaml . | nindent 8 }}
         {{- end }}
+        {{- if .Values.securityContext }}
+          {{- with .Values.securityContext }}
+            {{- toYaml . | nindent 8 }}
+          {{- end }}
```

Comment:
Since there is `.Values.nextcloud.podSecurityContext` now and `.Values.securityContext` is not documented anywhere, maybe just remove it?

Reply:
I can remove it, but the reason I kept it in was for some backwards compatibility.

Reply:
Yeah, but I wonder how many people actually rely on it since it's undocumented. Could you instead add a deprecation comment?

Reply:
I've added the deprecation comment to the `values.yaml` file.